Stripe/STPTelemetryClient.m

```diff
@@ -0,0 +1,188 @@
+//
+//  STPTelemetryClient.m
+//  Stripe
+//
+//  Created by Ben Guo on 4/18/17.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+@import WebKit;
+@import UIKit;
+#import <sys/utsname.h>
+
+#import "STPTelemetryClient.h"
+#import "STPAPIClient.h"
+
+typedef NS_ENUM(NSInteger, STPMetricField) {
+    STPMetricFieldCookieSupport = 0,
+    STPMetricFieldDoNotTrack,
+    STPMetricFieldLanguage,
+    STPMetricFieldPlatform,
+    STPMetricFieldPlugins,
+    STPMetricFieldScreenSize,
+    STPMetricFieldTimeZoneOffset,
+    STPMetricFieldTouchSupport,
+    STPMetricFieldAvailableStorage,
+    STPMetricFieldFonts,
+    STPMetricFieldGraphicsConfiguration,
+    STPMetricFieldUserAgent,
+    STPMetricFieldFlashVersion,
+    STPMetricFieldHasAdBlocker,
+    STPMetricFieldCanvasId,
+    STPMetricFieldMax
+};
+
+@interface STPTelemetryClient ()
+@property (nonatomic) NSDate *appOpenTime;
+@property (nonatomic, readwrite) NSURLSession *urlSession;
+@property (nonatomic, readwrite) NSString *userAgent;
+@property (nonatomic, readwrite) WKWebView *webView;
+@end
+
+@implementation STPTelemetryClient
+
++ (BOOL)shouldSendTelemetry {
+#if TARGET_OS_SIMULATOR
+    return NO;
+#endif
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunreachable-code"
+    return NSClassFromString(@"XCTest") == nil;
+#pragma clang diagnostic pop
+}
+
++ (instancetype)sharedInstance {
+    static id sharedClient;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedClient = [self new];
+    });
+    return sharedClient;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+        _urlSession = [NSURLSession sessionWithConfiguration:config];
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationDidBecomeActive) name:UIApplicationDidBecomeActiveNotification object:nil];
```

Comment:
I chose `didBecomeActive` because `willEnterForeground` doesn't get called on first launch. I know `didBecomeActive` can get called more often, but I think its behavior is ok for our purposes here.